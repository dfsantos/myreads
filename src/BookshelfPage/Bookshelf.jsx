import React from 'react';
import PropTypes from 'prop-types';

import Book from '../common/Book';

const propTypes = {
  books: PropTypes.array,
  name: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired,
  onCategorizeBook: PropTypes.func.isRequired,
};

const defaultProps = {
  books: [],
};

const byCategory = category => book => book.category === category;

function Bookshelf({ name, category, books, onCategorizeBook }) {
  return (
    <div className="bookshelf">
      <h2 className="bookshelf-title">{name}</h2>
      <div className="bookshelf-books">
        <ol className="books-grid">
          {books.filter(byCategory(category)).map(book => (
            <li key={book.id}>
              <Book
                title={book.title}
                coverLink={book.imageLinks.thumbnail}
                authors={book.authors}
                onChangeBookCategory={onCategorizeBook(book)}
              />
            </li>
          ))}
        </ol>
      </div>
    </div>
  );
}

Bookshelf.propTypes = propTypes;
Bookshelf.defaultProps = defaultProps;

export default Bookshelf;
