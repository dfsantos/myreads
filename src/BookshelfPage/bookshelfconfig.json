[
  { "name": "Currently Reading", "type": "currentlyReading" },
  { "name": "Read", "type": "read" },
  { "name": "Want to Read", "type": "wantToRead" }
]
