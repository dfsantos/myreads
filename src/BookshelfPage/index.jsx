import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import Bookshelf from './Bookshelf';

const propTypes = {
  books: PropTypes.array,
  onCategorizeBook: PropTypes.func.isRequired,
};

const defaultProps = {
  books: [],
};

class BookshelfPage extends Component {
  constructor(props) {
    super(props);
    this.onCategorizeBook = this.onCategorizeBook.bind(this);
  }

  onCategorizeBook(book) {
    return category => this.props.onCategorizeBook(Object.assign(book, { category }));
  }

  render() {
    const { books } = this.props;
    return (
      <div className="list-books">
        <div className="list-books-title">
          <h1>MyReads</h1>
        </div>
        <div className="list-books-content">
          <div>
            <Bookshelf
              name="Currently Reading"
              category="currentlyReading"
              books={books}
              onCategorizeBook={this.onCategorizeBook}
            />
            <Bookshelf
              name="Read"
              category="read"
              books={books}
              onCategorizeBook={this.onCategorizeBook}
            />
            <Bookshelf
              name="Want to Read"
              category="wantToRead"
              books={books}
              onCategorizeBook={this.onCategorizeBook}
            />
          </div>
        </div>
        <div className="open-search">
          <Link to="/search">Add a book</Link>
        </div>
      </div>
    );
  }
}

BookshelfPage.propTypes = propTypes;
BookshelfPage.defaultProps = defaultProps;

export default BookshelfPage;
