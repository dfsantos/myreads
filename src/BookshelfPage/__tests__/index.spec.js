import React from 'react';
import toJson from 'enzyme-to-json';

import BookshelfPage from '../index';

const onCategorizeBookListener = jest.fn(() => () => {});

it('should render without errors', () => {
  shallow(<BookshelfPage onCategorizeBook={onCategorizeBookListener} />);
});

it('snapshot without books', () => {
  const wrapper = shallow(<BookshelfPage onCategorizeBook={onCategorizeBookListener} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
