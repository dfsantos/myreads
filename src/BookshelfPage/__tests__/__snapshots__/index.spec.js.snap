// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot without books 1`] = `
<div
  className="list-books"
>
  <div
    className="list-books-title"
  >
    <h1>
      MyReads
    </h1>
  </div>
  <div
    className="list-books-content"
  >
    <div>
      <Bookshelf
        books={Array []}
        category="currentlyReading"
        name="Currently Reading"
        onCategorizeBook={[Function]}
      />
      <Bookshelf
        books={Array []}
        category="read"
        name="Read"
        onCategorizeBook={[Function]}
      />
      <Bookshelf
        books={Array []}
        category="wantToRead"
        name="Want to Read"
        onCategorizeBook={[Function]}
      />
    </div>
  </div>
  <div
    className="open-search"
  >
    <Link
      replace={false}
      to="/add-book"
    >
      Add a book
    </Link>
  </div>
</div>
`;
